File: src/Example1/Bank.java
package Example1;

import java.util.HashMap;
import java.util.Map;
import java.util.Random;
import java.util.Scanner;

public class Bank {
    private final Map<Long, Account> accounts;

    public Bank() {
        accounts = new HashMap<>();
    }

    public long createAccount() {
        Random random = new Random();
        long accountNumber = 3000000000L + (long) (random.nextDouble() * (6000000000L));
        Account account = new Account();
        account.setPin();
        account.getName();
        accounts.put(accountNumber, account);
        System.out.println("Account created successfully! Your account number is: " + accountNumber);
        return accountNumber;
    }

    public Account getAccount(long accountNumber) {
        if (accounts.containsKey(accountNumber)) {
            return accounts.get(accountNumber);
        } else {
            System.out.println("Account not found.");
            return null;
        }
    }

    public void deposit(long accountNumber, double amount) {
        Account account = getAccount(accountNumber);
        if (account != null) {
            account.deposit(amount);
        } else {
            System.out.println("Account not found.");
        }
    }

    public void withdraw(long accountNumber, double amount) {
        Account account = getAccount(accountNumber);
        if (account != null) {
            account.withdraw(amount);
        }
    }

    public void transfer(long fromAccountNumber, long toAccountNumber, String pin, double amount) {
        Account fromAccount = getAccount(fromAccountNumber);
        Account toAccount = getAccount(toAccountNumber);

        if (fromAccount != null && toAccount != null) {
            if (fromAccount.checkPin(pin)) {
                if (fromAccount.withdraw(amount)) {
                    toAccount.deposit(amount);
                    System.out.printf("Transfer of $%.2f from account %d to account %d successful.%n",
                            amount, fromAccountNumber, toAccountNumber);
                } else {
                    System.out.println("Insufficient funds.");
                }
            } else {
                System.out.println("Incorrect PIN for the source account.");
            }
        } else {
            System.out.println("One or both accounts not found.");
        }
    }

    public void checkAccountBalance(long accountNumber) {
        Account account = getAccount(accountNumber);
        if (account != null) {
            System.out.printf("Account Number: %d, Balance: $%.2f%n", accountNumber, account.checkBalance());
        }
    }

    public void checkAccountDetails(long accountNumber) {
        Account account = getAccount(accountNumber);
        if (account != null) {
            System.out.println(account.getName());
            System.out.println("Account Number: " + accountNumber);
            System.out.printf("Account Balance: $%.2f%n", account.checkBalance());
        }
    }

    public void lockAccount(long accountNumber) {
        Account account = getAccount(accountNumber);
        if (account != null) {
            account.lock();
            System.out.println("Account " + accountNumber + " locked successfully.");
        }
    }

    public void unlockAccount(long accountNumber) {
        Account account = getAccount(accountNumber);
        if (account != null) {
            account.unlock();
            System.out.println("Account " + accountNumber + " unlocked successfully.");
        }
    }

    public void addAccountNumber(long toAccountNumber) {
        if (!accounts.containsKey(toAccountNumber)) {
            accounts.put(toAccountNumber, new Account());
            System.out.println("Account " + toAccountNumber + " added successfully.");
        } else {
            System.out.println("Account already exists.");
        }
    }
}

File: dave_package/Bank.py
import random
from dave_package.Account import Account


class Bank:
    def __init__(self):
        self.accounts = {}

    def create_account(self):
        account_number = random.randint(3000000000, 9000000000)
        account = Account()
        account.set_pin()
        account.get_name()
        self.accounts[account_number] = account
        print(f"Account created successfully! Your account number is: {account_number}")
        return account_number

    def get_account(self, account_number):
        if account_number in self.accounts:
            return self.accounts[account_number]
        else:
            print("Account not found.")

    def deposit(self, account_number, amount):
        if account := self.get_account(account_number):
            account.deposit(amount)
        else:
            print("Account not found.")

    def withdraw(self, account_number, amount):
        if account := self.get_account(account_number):
            account.withdraw(amount)

    def transfer(self, from_account_number, to_account_number, pin, amount):
        if from_account_number in self.accounts and to_account_number in self.accounts:
            from_account = self.accounts[from_account_number]
            if from_account.check_pin(pin):
                to_account = self.accounts[to_account_number]

                if from_account.withdraw(amount):
                    to_account.deposit(amount)
                    print(
                        f"Transfer of ${amount:.2f} from account {from_account_number} to account {to_account_number} successful.")
            else:
                print("Incorrect PIN for the source account.")
        else:
            print("One or both accounts not found.")

    def check_account_balance(self, account_number):
        if account := self.get_account(account_number):
            balance = account.check_balance()
            print(f"Account Number: {account_number}, balance: ${balance:.2f}")
        else:
            print("Account not found.")

    def check_account_details(self, account_number):
        if account := self.get_account(account_number):
            print(f"Account Number: {account_number}")
            print(f"Account Holder: {account.get_name()}")
            print(f"Account Balance: ${account.check_balance():.2f}")
        else:
            print("Account not found.")

    def lock_account(self, account_number):
        if account := self.get_account(account_number):
            account.lock()
            print(f"Account {account_number} locked successfully.")
        else:
            print("Account not found.")

    def unlock_account(self, account_number):
        if account := self.get_account(account_number):
            account.unlock()
            print(f"Account {account_number} unlocked successfully.")
        else:
            print("Account not found")

    def add_account_number(self, to_account_number):
        if to_account_number not in self.accounts:
            self.accounts[to_account_number] = Account()
            print(f"Account {to_account_number} added successfully.")
        else:
            print("Account already exists.")
